### PR TITLE
docs: added environment variables documentation

### PR DESCRIPTION
**Pull request by HariSekhon**
I'd like an environment variables documentation reference page.

It's the sort of thing I use a lot in other projects documentation so I've created one.

I've added a documentation page on some useful environment variables that I add a lot as I've pieced these together over time from other pages, this is slightly different to the build environment page, but feel free to incorporate or restructure this as you see fit.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks for the contribution, and apologies for the slow review! Added one nitpick.

File: docs/user-guide/environment-variables.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Environment Variables
+
+The following environment variables can be used with `argocd` CLI:
+
+| Environment Variable | Description |
+| --- | --- |
+| `ARGOCD_SERVER` | the address of the ArgoCD server without `https://` prefix <br> (instead of specifying `--server` for every command) <br> eg. `ARGOCD_SERVER=argocd.mycompany.com` if served through an ingress with DNS |
+| `ARGO_AUTH_TOKEN` | the ArgoCD `apiKey` for your ArgoCD user to be able to authenticate |
```
[the remark this answers is not in the document]
Comment (HariSekhon) [reply docs/user-guide/environment-variables.md]: Yes well spotted, have updated the typo, thanks.

**Comment by lucasff** [comment]
this is ready, no? is there any way to passing the username or password via env?

**Comment by HariSekhon** [comment]
@lucasff you got me curious, so I've just grep'd the entire ArgoCD codebase for tokens using a new script I just rustled up:

https://github.com/HariSekhon/DevOps-Bash-tools/blob/master/git_grep_env_vars.sh

This has led to me finding some new environment variables:

- ARGOCD_CONFIG_DIR
- XDG_CONFIG_HOME

and some other interesting bits:

```
pkg/apiclient/apiclient.go: EnvArgoCDAuthToken = "ARGOCD_AUTH_TOKEN"
pkg/apiclient/apiclient.go: EnvArgoCDServer = "ARGOCD_SERVER"
pkg/apiclient/apiclient.go: EnvArgoCDgRPCMaxSizeMB = "ARGOCD_GRPC_MAX_SIZE_MB"
...

common/common.go:   EnvAppConfigPath = "ARGOCD_APP_CONF_PATH"
common/common.go:   EnvControllerReplicas = "ARGOCD_CONTROLLER_REPLICAS"
common/common.go:   EnvControllerShard = "ARGOCD_CONTROLLER_SHARD"
common/common.go:   EnvEnableGRPCTimeHistogramEnv = "ARGOCD_ENABLE_GRPC_TIME_HISTOGRAM"
common/common.go:   EnvGitAttemptsCount = "ARGOCD_GIT_ATTEMPTS_COUNT"
common/common.go:   EnvGitRetryDuration = "ARGOCD_GIT_RETRY_DURATION"
common/common.go:   EnvGitRetryFactor = "ARGOCD_GIT_RETRY_FACTOR"
common/common.go:   EnvGitRetryMaxDuration = "ARGOCD_GIT_RETRY_MAX_DURATION"
common/common.go:   EnvGitSubmoduleEnabled = "ARGOCD_GIT_MODULES_ENABLED"
common/common.go:   EnvGithubAppCredsExpirationDuration = "ARGOCD_GITHUB_APP_CREDS_EXPIRATION_DURATION"
common/common.go:   EnvGnuPGHome = "ARGOCD_GNUPGHOME"
common/common.go:   EnvHelmIndexCacheDuration = "ARGOCD_HELM_INDEX_CACHE_DURATION"
common/common.go:   EnvLogFormat = "ARGOCD_LOG_FORMAT"
common/common.go:   EnvLogLevel = "ARGOCD_LOG_LEVEL"
common/common.go:   EnvMaxCookieNumber = "ARGOCD_MAX_COOKIE_NUMBER"
common/common.go:   EnvPauseGenerationAfterFailedAttempts = "ARGOCD_PAUSE_GEN_AFTER_FAILED_ATTEMPTS"
common/common.go:   EnvPauseGenerationMinutes = "ARGOCD_PAUSE_GEN_MINUTES"
common/common.go:   EnvPauseGenerationRequests = "ARGOCD_PAUSE_GEN_REQUESTS"
common/common.go:   EnvPluginSockFilePath = "ARGOCD_PLUGINSOCKFILEPATH"
common/common.go:   EnvVarRBACDebug = "ARGOCD_RBAC_DEBUG"
common/common.go:   EnvVarSSHDataPath = "ARGOCD_SSH_DATA_PATH"
common/common.go:   EnvVarSSODebug = "ARGOCD_SSO_DEBUG"
common/common.go:   EnvVarTLSDataPath = "ARGOCD_TLS_DATA_PATH"
common/common.go:   EnvWatchAPIBufferSize = "ARGOCD_WATCH_API_BUFFER_SIZE"
...

util/cache/cache.go:    envRedisPassword = "REDIS_PASSWORD"
...

reposerver/cache/cache.go:  cmd.Flags().DurationVar(&repoCacheExpiration, "repo-cache-expiration", env.ParseDurationFromEnv("ARGOCD_REPO_CACHE_EXPIRATION", 24*time.Hour, 0, math.MaxInt64), "Cache expiration for repo state, incl. app lists, app details, manifest generation, revision meta-data")
reposerver/cache/cache.go:  cmd.Flags().DurationVar(&revisionCacheExpiration, "revision-cache-expiration", env.ParseDurationFromEnv("ARGOCD_RECONCILIATION_TIMEOUT", 3*time.Minute, 0, math.MaxInt64), "Cache expiration for cached revision")
...

cmd/argocd-repo-server/commands/argocd_repo_server.go:  command.Flags().BoolVar(&disableTLS, "disable-tls", env.ParseBoolFromEnv("ARGOCD_REPO_SERVER_DISABLE_TLS", false), "Disable TLS on the gRPC endpoint")
cmd/argocd-repo-server/commands/argocd_repo_server.go:  command.Flags().Int64Var(&parallelismLimit, "parallelismlimit", int64(env.ParseNumFromEnv("ARGOCD_REPO_SERVER_PARALLELISM_LIMIT", 0, 0, math.MaxInt32)), "Limit on number of concurrent manifests generate requests. Any value less the 1 means no limit.")
cmd/argocd-repo-server/commands/argocd_repo_server.go:  command.Flags().StringVar(&cmdutil.LogFormat, "logformat", env.StringFromEnv("ARGOCD_REPO_SERVER_LOGFORMAT", "text"), "Set the logging format. One of: text|json")
cmd/argocd-repo-server/commands/argocd_repo_server.go:  command.Flags().StringVar(&cmdutil.LogLevel, "loglevel", env.StringFromEnv("ARGOCD_REPO_SERVER_LOGLEVEL", "info"), "Set the logging level. One of: debug|info|warn|error")
```

 among other environment variables too many to list here.

This is also:

```
test/e2e/fixture/fixture.go:    EnvAdminPassword = "ARGOCD_E2E_ADMIN_PASSWORD"
```
but that looks like something only for tests and not what you want, so @lucasff  think you'll have to stick with `ARGOCD_AUTH_TOKEN`.

Since I don't have a thorough experience with all the environment variables, I'd rather limit this PR to the ones I know and merge this so users can actually find it on the website.

**Comment by crenshaw-dev** [comment]
@HariSekhon yeah documenting the env vars is tricky, because where they get used depends a lot on context. Many of these are specific to running the CLI to start an api server or controller or repo server. 

This is a good start, 'cause it's focused on env vars that generically make sense for all CLI operations.

**Review by crenshaw-dev — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
Cherry-picked onto 2.4 (I'm trying to be more aggressive about cherry-picking docs-only PRs - they're useful and virtually zero risk.)

**Comment by lucasff** [comment]
@HariSekhon thanks a lot for doing that! I ended up using manually assigned env vars to run `argocd login` on my GitHub Actions workflows ;)

**Comment by HariSekhon** [comment]
@lucasff you're welcome. Btw if it's public, please post a link as I'm always interested in seeing how other people solve CI/CD as I do a lot of this work myself too, and have a couple repos you might find interesting for [Jenkins](https://github.com/HariSekhon/Jenkins) and [GitHub-Actions](https://github.com/HariSekhon/GitHub-Actions) as well.

**Pull request merged**